### PR TITLE
Extract deceased field to its own function

### PR DESCRIPTION
**Pull request by anthonybaulo**
Types of changes
----------------

- Code cleanup / Refactoring

Description
-----------

Extract deceased field to its own function with tests.
Rename previous parse_deceased_field() to process_deceased_field()

Checklist:
----------

-  [x] I have updated the documentation accordingly
-  [x] I have written unit tests

Fixes: scrapd/scrapd#106

### CODE REVIEW

**Comment by anthonybaulo** [comment]
@rgreinho 
There's a lot going on with parsing the deceased field (L344-470 in `apd.py`), including a function named `parse_deceased_field()`. 
Is this issue solely concerned with extracting a parsing function for use inside of `parse_page_content()` on L504, which would literally just be moving the existing regex from the `searches` list into a new function?

```python
searches = [
        (Fields.DECEASED, re.compile(r'>Deceased:\s*(?:</span>)?(?:</strong>)?\s*>?([^<]*\d)\s*.*\)?<')),
        (Fields.LOCATION, re.compile(r'>Location:.*>\s{2,}(?:</strong>)?([^<]+)')),
    ]
```

If so, what would you like to call the new function, since `parse_deceased_field()` is being used for something else?

**Comment by rgreinho** [comment]
Oh you're right! so let's do the following:

* Rename the existing `parse_deceased_field()` function to `process_deceased_field()`. Be sure to catch all the references to this functions. VS Code will be your friend :)
* Name the new function you will create `parse_deceased_field()`.

And we should be good to go!

File: scrapd/core/apd.py [hunk 1]
```diff
@@ -533,6 +532,9 @@ def parse_page_content(detail_page, notes_parsed=False):
     if date_field_str:
         d[Fields.DATE] = date_utils.parse_date(date_field_str)
 
+    # Parse the `Deceased` field.
+    d[Fields.DECEASED] = parse_deceased_field(normalized_detail_page)
```
Comment (rgreinho) [inline scrapd/core/apd.py]: Apply the same pattern as for the date field (l531-532). You can also look at my other PR which does something similar: https://github.com/scrapd/scrapd/pull/138/files#diff-2de3abbf558189989f23bce31a4003f1R528.

**Comment by rgreinho** [comment]
And merged! Great work @anthonybaulo!

**Pull request merged**